compiler/resolution/src/org/jetbrains/kotlin/resolve/calls/tower/ScopeTowerProcessors.kt

```diff
@@ -71,8 +72,12 @@ internal abstract class AbstractSimpleScopeTowerProcessor<C : Candidate>(
     ): Collection<C> {
         val result = mutableListOf<C>()
         for (candidate in collector) {
-            if (candidate.requiresExtensionReceiver == (extensionReceiver != null) &&
-                candidate.descriptor.additionalReceiverParameters.size == additionalReceivers.size
+            val hasExtensionReceiver = extensionReceiver != null
+            val descriptorAdditionalReceiversSize = candidate.descriptor.additionalReceiverParameters.size
+            val callingClassConstructorWithAdditionalReceiver = candidate.descriptor is ClassConstructorDescriptor && hasExtensionReceiver
+                    && !candidate.requiresExtensionReceiver && descriptorAdditionalReceiversSize == 1
```

Comment:
Maybe, it'd be simpler if we passed additional receivers for constructor within `additionalReceivers` parameter

Reply:
But if it's not easy now, I don't insist

Reply:
Tried both options and for now from the point of view of backend implementation it's easier to leave the current one.

Reply:
Since the backend is in progress, most likely I will adjust this in the next PR.